tools/api-docs-generator/changelog/changelog.go

```diff
@@ -0,0 +1,255 @@
+
+	allVersions, err := versions.GetCurrentVersions(ctx, cfg)
+	if err != nil {
+		return "", err
+	}
+
+	latestGAVersion := versions.GetLatestGAVersion(allVersions)
+
+	historicalChangelog, err := os.ReadFile(changeLogFileName) // just pass the file name
+	if err != nil {
+		fmt.Print(err)
+	}
+
+	writer, err := os.Create(changeLogFileName)
+	if err != nil {
+		return "", err
+	}
+
+	defer func(writer *os.File) {
+		writeErr := writer.Close()
+		if writeErr != nil {
+			fmt.Printf("Error closing writer for %s\n", changeLogFileName)
+		}
+	}(writer)
+
+	nextURL := fmt.Sprintf("%s/%s", cfg.Fetcher.Source, latestGAVersion)
+	baseURL := cfg.Fetcher.Destination
+
+	groupedChanges, err := getChangeLog(nextURL, baseURL, loader)
+	if err != nil {
+		return "", err
+	}
+
+	if len(groupedChanges) == 0 {
+		return "", nil
+	}
+
+	markdown := md.NewMarkdown(writer)
+
+	err = WriteToChangeLog(markdown, groupedChanges, latestGAVersion, nextURL, syncStateCfg.LastSyncedVersion)
+	if err != nil {
+		return "", err
+	}
+
+	err = markdown.Build()
+	if err != nil {
+		return "", err
+	}
+
+	_, err = writer.Write(historicalChangelog)
+	if err != nil {
+		return "", err
+	}
+
+	return latestGAVersion, err
+}
+
+func GenerateHistorical(ctx context.Context, cfg *config.Config, changeLogFileName, endVersion string) error {
+	allVersions, err := versions.GetCurrentVersions(ctx, cfg)
+	if err != nil {
+		return err
+	}
+
+	loader := openapi3.NewLoader()
+	loader.IsExternalRefsAllowed = true
+
+	gaVersions := versions.ExtractGAVersions(allVersions)
+	endVersionPos := sort.SearchStrings(gaVersions, endVersion)
+	gaVersions = gaVersions[:endVersionPos]
+	slices.Reverse(gaVersions)
+	nextVersion := gaVersions[0]
+
+	writer, err := os.Create(changeLogFileName)
+	if err != nil {
+		return err
+	}
+
+	for _, baseVersion := range gaVersions[1:] {
+		nextURL := fmt.Sprintf("%s/%s", cfg.Fetcher.Source, nextVersion)
+		baseURL := fmt.Sprintf("%s/%s", cfg.Fetcher.Source, baseVersion)
+		groupedChanges, err := getChangeLog(nextURL, baseURL, loader)
+		if err != nil {
+			return err
+		}
+
+		groupedChanges = filterChanges(groupedChanges)
+
+		if len(groupedChanges) != 0 {
+			markdown := md.NewMarkdown(writer)
+
+			err = WriteToChangeLog(markdown, groupedChanges, nextVersion, nextURL, "")
+			if err != nil {
+				return err
+			}
+
+			err = markdown.Build()
+			if err != nil {
+				return err
+			}
+		}
+		nextVersion = baseVersion
+	}
+
+	return writer.Close()
+}
+func filterChanges(changes []ChangesByEndpoint) []ChangesByEndpoint {
+	filteredChanges := make([]ChangesByEndpoint, 0, len(changes))
+	for _, endpointChange := range changes {
+		if strings.HasPrefix(endpointChange.Path, "/openapi") {
+			continue
+		}
+		filteredChanges = append(filteredChanges, endpointChange)
+	}
+	return filteredChanges
+}
+
+func WriteToChangeLog(markdown *md.Markdown, groupedChanges []ChangesByEndpoint, baseVersion, nextVersionURL, lastSyncVersion string) error {
```

Comment:
```suggestion
func writeToChangeLog(markdown *md.Markdown, groupedChanges []ChangesByEndpoint, baseVersion, nextVersionURL, lastSyncVersion string) error {
```
This function could be a package level one